api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -1982,6 +1984,72 @@ public boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType)
 		return dao.getConceptAttributeCount(conceptAttributeType) > 0;
 	}
 
+	/**
+	 * @see org.openmrs.api.ConceptService#getConceptByNameOrIdOrMap(String conceptRef
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Concept getConceptByNameOrIdOrMap(String conceptRef) throws APIException {
+		if (conceptRef == null) {
+			return null;
+		}
+		Concept cpt = null;
+		//check if input is a valid Uuid
+		if (!(conceptRef.length() < 36 || conceptRef.length() > 38 || conceptRef.contains(" ") || conceptRef.contains("."))) {
```

Comment:
should it be greater than 36 or 38?

Reply:
Ugh... I hate the 36 / 38 thing. The normal display format for UUIDs is 32 digits + 4 dashes (36 characters), e.g. 8-4-4-4-12. However, our database fields allow UUIDs to be up to 38 characters (which corresponds to the Microsoft GUID format, which is the same as the above with curly braces around it, even though we don't use that anywhere).

Maybe this could be simplified into just checking with a regex:

```java
private static final Pattern UUID_FORMAT = Pattern.compile("^\\d{8}-\\d{4}-\\d{4}-\\d{4}-\\d{12}$");

...

if (UUID_FORMAT.matcher(conceptRef).matches()) {
    ...
}
```

Reply:
Wow, Thank you Ivan!  The information you have given has even made it understand it the more

Reply:
Hello @ibacher @HerbertYiga Thanks for the support.  
I tried to implement @ibacher's suggestion of using a simpler UUID check  but i failed because i got some conflicts which i tried to solve, but failed.  
How i wish we leave the UUID as it was.  siince i have commented, it will be understood by all those that will use the method.  
cc @dkayiwa @samuelmale